### PR TITLE
ci(azure): add pipeline for testing

### LINKED ISSUE #49: "yarn run generate" fails
Tried building on my Windows machine but I didn't have Python installed which is required for node-gyp. I then tried building it in a Linux environment with Python but it failed with TypeScript errors. I guess the `lint` command doesn't find these errors so adding a build task to CI might prevent these issues.

<details><summary>Logs</summary>

```
$ nuxt-ts generate
[fatal] Nuxt build error
  ERROR in mixins/imageHelper.ts:10:22
  TS2339: Property '$axios' does not exist on type '{ getImageLink(id: string, type: string): string; }'.
  8 |      */
  9 |     getImageLink(id: string, type: string): string {
  > 10 |       return `${this.$axios.defaults.baseURL}/Items/${id}/Images/${type}`;
  |                      ^^^^^^
  11 |     }
  12 |   }
  13 | };
  
  ERROR in pages/item/_itemId.vue:37:17
  TS2533: Object is possibly 'null' or 'undefined'.
  35 |     });
  36 |
  > 37 |     this.item = Item.data.Items[0];
  |                 ^^^^^^^^^^^^^^^
  38 |   }
  39 | });
  40 | </script>
   ╭─────────────────────────────╮
   │                             │
   │   ✖ Nuxt Fatal Error        │
   │                             │
   │   Error: Nuxt build error   │
   │                             │
   ╰─────────────────────────────╯
```

</details>

### PR DESCRIPTION
**Pull request by Camc314**
adds `yarn build` to pipeline

Fixes resulting errors

Fixes #49

### CODE REVIEW

File: pages/item/_itemId.vue [hunk 1]
```diff
@@ -16,25 +16,32 @@
 
 <script lang="ts">
 import Vue from 'vue';
+import { BaseItemDto } from '~/api';
 import imageHelper from '~/mixins/imageHelper';
 
+interface itemsArray extends Array<BaseItemDto> {
```
Comment (MrTimscampi) [inline pages/item/_itemId.vue:22]: You can use `BaseItemDto[]` for this

File: mixins/imageHelper.ts [hunk 2]
```diff
@@ -1,4 +1,11 @@
-const imageHelper = {
+/**
+ * This server as a helper for geting image urls from the itemId
```
Comment (MrTimscampi) [inline mixins/imageHelper.ts:2]:
```suggestion
 * Helper for image manipulation and image-related utility functions
```

File: mixins/imageHelper.ts [hunk 3]
```diff
@@ -1,4 +1,11 @@
...
+ *
+ * @mixin
+ */
+import Vue from 'vue';
+
+const imageHelper = Vue.extend({
   methods: {
     /**
      * returns a URL with the link to the image
```
Comment (MrTimscampi) [inline mixins/imageHelper.ts:11]:
```suggestion
     * Get the URL of an item's image
```

File: .ci/azure-pipelines-build.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+          key: 'yarn | yarn.lock'
+          path: 'node_modules'
+          cacheHitVar: CACHE_RESTORED
+
+      - script: 'yarn install --frozen-lockfile'
+        displayName: 'Install Dependencies'
+        condition: ne(variables.CACHE_RESTORED, 'true')
+
+      - script: 'yarn build'
+        displayName: 'Building'
```
Comment (MrTimscampi) [inline .ci/azure-pipelines-build.yml:26]:
```suggestion
        displayName: 'Build'
```

**Review by MrTimscampi — CHANGES_REQUESTED** [review]

File: pages/item/_itemId.vue (hunk 1 above)
Comment (Camc314) [reply pages/item/_itemId.vue]: Good to know!

**Review by MrTimscampi — APPROVED** [review]

**Pull request merged**